File: Lab3/Lab(C++)/classes.cpp
//
// Created by Danya Butov on 17.04.2022.
//

#include "classes.h"

// Методи класса тривимірного вектора

// Конструктор класса
ThreeDimensionalVector::ThreeDimensionalVector(int x_value, int y_value, int z_value) {
    x = x_value;
    y = y_value;
    z = z_value;
    set_length();
}

// Отримати координатори вектора
void ThreeDimensionalVector::get_values() const {
    cout << "x= " << x << "; y= " << y << "; z= " << z << ";" << endl;
}

// Встановити значення довжини вектора
void ThreeDimensionalVector::set_length() {
    length = sqrt(pow(x, 2) + pow(y, 2) + pow(z, 2));
}

// Отримати значення довжини
double ThreeDimensionalVector::get_length() const {
    return length;
}

File: Lab3/Lab(Python)/classes.py
import math


# Create a new class "Third Dimensional Vector"
class ThirdDimensionalVector:
    length = 0

    def __init__(self, x_value, y_value, z_value):
        self.__x = x_value
        self.__y = y_value
        self.__z = z_value
        self._set_length()

    def _set_length(self):
        self.length = math.sqrt(pow(self.__x, 2) + pow(self.__y, 2) + pow(self.__z, 2))

    def get_values(self):
        string = "x -> {x}; y -> {y}; z -> {z}"
        return string.format(x=self.__x, y=self.__y, z=self.__z)

    def get_length(self):
        return round(self.length, 2)